### PR TITLE
Update to rails 7

### PR DESCRIPTION
**Pull request by noplisu**
## Pull Request Summary

To keep things up to date we want to update to rails 7.

## Description of the Changes

Here we update the gem version in the gem file, run the `rails app:update` and clean up the generated code.

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -1,4 +1,3 @@
 Rails.application.config.assets.version = '1.0'
-Rails.application.config.assets.paths << Rails.root.join('node_modules')
 Rails.application.config.assets.precompile += %w(application.css application.js bootstrap.min.js popper.js)
-# Rails.application.config.assets.unknown_asset_fallback = true
+Rails.application.config.assets.unknown_asset_fallback = true
```
Comment (torrocus) [inline config/initializers/assets.rb:3]: Is it ok for us to have this fallback enabled?

File: config/environments/production.rb [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+  config.action_controller.allow_forgery_protection = false
   config.action_controller.perform_caching = true
   config.action_mailer.perform_caching = false
   config.active_record.dump_schema_after_migration = false
   config.active_storage.service = :local
   config.active_support.deprecation = :notify
+  config.active_support.report_deprecations = false
   config.assets.compile = false
   config.assets.css_compressor = :sass
   # config.assets.js_compressor = :uglifier
```
Comment (torrocus) [inline config/environments/production.rb:13]: It's not relevant to this task, but in general I'm wondering if we shouldn't enable js compressor in production. :thinking:

File: config/initializers/assets.rb (hunk 1 above)
Comment (noplisu) [reply config/initializers/assets.rb:3]: true is the default value. So both lines are equivalent. Setting it to true is just more direct.

File: config/environments/production.rb (hunk 2 above)
Comment (noplisu) [reply config/environments/production.rb:13]: It would probably be better. I didn't do anything with it since it was out of scope

**Review by torrocus — APPROVED** [review]

**Pull request merged**